### PR TITLE
completed user story and edge case requirements for Airport challenge

### PR DESCRIPTION
**Pull request by loushark**

### CODE REVIEW

File: lib/airport.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+require_relative 'planes'
+require_relative 'weather'
+
+class Airport
+
+  attr_reader :apron, :capacity, :weather
```
Comment (louballantyne) [inline lib/airport.rb:6]: In the case that there are 10 planes at the airport and somebody tries to change the capacity to 1, are they able to do this, or does the code stop them?

File: lib/airport.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  def initialize(capacity = 1)
+    @apron = []
+    @capacity = capacity
+    @weather = Weather.new
+  end
+
+  def land(plane)
+    fail "A plane can't land, there's no room!" if apron.length == capacity
```
Comment (louballantyne) [inline lib/airport.rb:15]: Maybe a test of the airport being at capacity could be a method?

File: lib/airport.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    fail "This plane is already at another airport" if
+    plane.status == "landed"
+
+    apron << plane.name
+    plane.status = "landed"
+    "#{plane.name} has landed!"
+    # change_status(plane)
+  end
+
+  def take_off(plane)
```
Comment (louballantyne) [inline lib/airport.rb:31]: Can planes take off during bad weather?

File: spec/weather_spec.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+require 'weather'
+require 'airport'
+
+describe Weather do
+  it "returns a random number" do
+    expect(subject).to receive(:rand)
+    subject.weather_report
+  end
```
Comment (louballantyne) [inline spec/weather_spec.rb:8]: Could you also test whether the weather reports itself as stormy or sunny?

File: spec/airport_spec.rb [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    colin = Plane.new('colin')
+    expect { heathrow.land(colin) }.to raise_error("A plane can't land, there's no room!")
+  end
+
+  it "allows airports to override the default capacity" do
+    expect(heathrow.capacity).to eq(1)
+    gatwick = Airport.new(5)
+    expect(gatwick.capacity).to eq(5)
+  end
+
```
Comment (louballantyne) [inline spec/airport_spec.rb:48]: Can a user change the capacity of the airport while the program is running?

File: lib/airport.rb [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+    return fail "Plane is not at this airport" unless apron.include?(plane.name)
+
```
Comment (louballantyne) [inline lib/airport.rb:34]: This is a really nice, catch-all way to check if the plane is valid for take off.

File: lib/weather.rb [hunk 7]
```diff
@@ -0,0 +1,9 @@
+class Weather
+  def initialize
+  end
+
+  def weather_report
+    # where every number is sunny except for 6 which is a rare occasion of stormy
+    rand(1..9) == 6 ? "stormy" : "sunny"
+  end
```
Comment (louballantyne) [inline lib/weather.rb:8]: Great use of the ternary operator here!

File: README.md [hunk 8]
```diff
@@ -60,30 +43,24 @@ To ensure safety
...
+To feature test in irb:  
+> require './lib/airport'  
+  
+example instances to test:  
+> heathrow = Aiport.new (default capacity is 1)  
+> gatwick = Airport.new(10) (default capacity overriden to be 10)  
+> bertie = Plane.new('Bertie')  
+> colin = Plane.new('Colin')  
+> heathrow.land(bertie) => "Bertie has landed!" (provided weather is sunny)  
+> heathrow.land(colin) => "A plane can't land, there's no room!" (heathrow only has a capacity of 1, bertie is already on the apron)
```
Comment (louballantyne) [inline README.md:65]: I love these examples - it's really helpful to understand how the code works.

**Review by louballantyne — COMMENTED** [review]
You named your methods and variables clearly, making it easy to understand the code. I think it could possibly be refactored slightly so that things are extracted to additional methods. Well done on the full test coverage!

File: lib/airport.rb (hunk 1 above)
Comment (loushark) [reply lib/airport.rb:6]: The capacity is set when the airport instance is created so the capacity shouldn't be able to change. Useful to write an extra test to ensure this cannot be changed though

File: spec/airport_spec.rb (hunk 5 above)
Comment (loushark) [reply spec/airport_spec.rb:48]: Shouldn't be able to. The capacity is only readable and set during Airport.new.

**Pull request closed without merging**